Homework6/ConsoleApp1/Classes/Logger.cs

```diff
@@ -0,0 +1,72 @@
+﻿using System.Text;
+using System.Reflection;
+using System.Text.Json;
+using Task1.Attributes;
+
+namespace Task1.Classes
+{
+ public class Logger
+ {
+  public string JSONName { get; set; }
+
+  public Logger(string name) => JSONName = name;
+
+  // Not sure if this is exactly what the task meant
+  public void Track<T>(params T[] objects)
+  {
+   List<PropertyInfo> GetTrackingProperties<T>()
+   {
+    List<PropertyInfo> list = new List<PropertyInfo>();
+    PropertyInfo[] propertyInfos = typeof(T).GetProperties();
+   
+    foreach (var prop in propertyInfos)
+    {
+     object[] attributes = prop.GetCustomAttributes(true);
+     foreach (var attr in attributes)
+     {
+      if (attr is TrackingPropertyAttribute)
+      {
+       list.Add(prop);
+      }
+     }
+    }
+    return list;
+   }
+
+   string GetSerializedObjectStringByPropertyList<T>(List<PropertyInfo> list, params T[] objects)
+   {
+    StringBuilder sb = new StringBuilder();
+
+    foreach (var obj in objects)
+    {
+     if (typeof(T).IsDefined(typeof(TrackingEntityAttribute), false))
+     {
+      foreach (var property in list)
+      {
+       if (property is not null)
+       {
+        sb.Append(property.Name);
```

Comment:
> Values are captured as
> "name:value" pairs, where "name" is taken from the [TrackingProperty] attribute. If
> the name is not specified there (empty), then the name of the property or field to which
> the [TrackingProperty] attribute is applied is taken.

name should be  value of  TrackingProperty.PropertyName attribute. 
Only when it is empty or null, then original property name is used